libsel4/include/sel4/macros.h

```diff
@@ -43,6 +43,14 @@
 
 #endif
 
+/* An expression that does nothing, but it is still a token that requires a
+ * semicolon afterwards. Use this for macros that are reduced to nothing in
+ * certain configurations, so the semicolon is properly consumed and does not
+ * create side effects in corner cases. Using "do {} while(0)" here is another
+ * common method, but "((void)(0))" is preferred for verification.
```

Comment:
Is it? I have the feeling it's the first time I see that expression.

I suspect the proof failures are because of it. I'll have to look at what this translates to in the semantics, but I suspect it'll be easy enough to make a rewrite rule that turns this into SKIP, so it's probably fine to use. I don't think it's substantially different to `do {} while (0)` for verification (which probably also will need its own rule).

Reply:
The other reason the proofs might be failing (the error message in the log doesn't show enough info) is that `wordRadix` isn't used in the expression any more, which means there is less simplification necessary, which means means the proofs steps might change slightly.

I still don't really mind either way. But as it stands it'll need a small proof update. I can look at that if needed.